workflows/ampliseq.nf

```diff
@@ -44,12 +44,24 @@ if (params.classifier) {
 
 if (params.dada_ref_taxonomy && !params.skip_taxonomy) {
     ch_dada_ref_taxonomy = Channel.fromList(params.dada_ref_databases[params.dada_ref_taxonomy]["file"]).map { file(it) }
+    if (params.addsh) {
+        if (params.dada_ref_databases[params.dada_ref_taxonomy]["shfile"]) {
+            ch_shinfo = Channel.fromList(params.dada_ref_databases[params.dada_ref_taxonomy]["shfile"]).map { file(it) }
+        } else {
+            exit 1, "SH information is not available for current selection of reference database"
+        }
```

Comment:
This could be moved to [lib/WorkflowAmpliseq.groovy](https://github.com/nf-core/ampliseq/blob/master/lib/WorkflowAmpliseq.groovy) where typically parameter validation is taken care of.
That should also stop the workflow early so that the parameters are revisited.
Also, I think it would be nice to mention here the relevant parameters `--dada_ref_taxonomy` & `--addsh` and highlighting that this only works with UNITE dbs.

Reply:
Thanks for pointing me to that file! Moving the code...